lib/topological_inventory/schema/base.rb

```diff
@@ -18,26 +21,28 @@ def to_hash
         }.merge(super)
       end
 
-      def self.klass_for(name)
-        klass_name = "TopologicalInventory::Schema::#{name}"
-        klass = klass_name.safe_constantize
-
-        # sometimes autoloader is confused and loads class with another namespace
-        # checks whether it loads what we want
-        klass if klass.to_s == klass_name
-      end
+      class << self
+        def from_hash(persister_data, manager)
+          persister = super(persister_data, manager)
+          %i[
+            refresh_state_part_collected_at
+            refresh_state_part_sent_at
+            refresh_state_started_at
+            refresh_state_sent_at
+            ingress_api_sent_at
+          ].each do |timestamp_name|
+            persister.refresh_time_tracking[timestamp_name] = persister_data[timestamp_name.to_s]
```

Comment:
This feels like it belongs in the inventory_refresh gem [here](https://github.com/ManageIQ/inventory_refresh/blob/master/lib/inventory_refresh/persister.rb#L140-L145)

Reply:
I also tought about it, but this tracking seems to be topological-inventory oriented...but :+1:

Reply:
Yeah that version of inventory_refresh is all topo specific already, and when we bring that back to manageiq it will be the whole topo system (persisters and ingress-api)